lib/logstash/filters/kv.rb

```diff
@@ -238,21 +239,14 @@ def filter(event)
     # Add default key-values for missing keys
     kv = @default_keys.merge(kv)
 
-    # If we have any keys, create/append the hash
-    if kv.length > 0
-      if @target.nil?
-        # Default is to write to the root of the event.
-        dest = event.to_hash
+    unless kv.empty?
```

Comment:
minor readability suggestion:

instead of 
```ruby
unless kv.empty?
  if @target
    @logger.debug? && @logger.debug("Overwriting existing target field", :target => @target)
    event[@target] = kv
  else
    kv.each{|k, v| event[k] = v}
  end

  filter_matched(event)
end
```

reduce 1 level with:

```ruby
return if kv.empty?

if @target
  @logger.debug? && @logger.debug("Overwriting existing target field", :target => @target)
  event[@target] = kv
else
  kv.each{|k, v| event[k] = v}
end

filter_matched(event)
```